### PR TITLE
Feature/GitHub action to build wheel

### PR DESCRIPTION
**Pull request by stephanie-mak**

### CODE REVIEW

File: commit.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+set -ex
+
+git config --global user.name 'github action'
+
+if [ -n "$(git status --porcelain)" ]; then
+	echo "there are changes";
+	git add package/*
+	git commit -m 'publish wheel by github action'
+	git push
```
Comment (mao-liu) [inline commit.sh:11]: hey @stephanie-mak , does this avoid the infinite loop, where the commit and push would trigger github action to run again, which will then make another build/commit/push?

File: commit.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -ex
+
+git config --global user.name 'github action'
+
```
Comment (mao-liu) [inline commit.sh:6]:
```suggestion
if [ $(git log -n 1 --format=format:%an) -eq "github action" ]; then
    echo "previous commit is by github action, skipping commit"
    exit 0
fi
```

File: commit.sh (hunk 1 above)
Comment (stephanie-mak) [reply commit.sh:14]: there might be a lock in github action since infinite loop behaviour is not observed in the data asset repo

**Comment by mao-liu** [comment]
Hey @stephanie-mak , upon further thought I think we should use the GitHub Releases feature to track release artifacts, rather than committing the built artifact into git.

I think we will need to incorporate the workflow from here: https://github.com/marketplace/actions/publish-release

I will try to update the action spec when I have some time!

**Comment by stephanie-mak** [comment]
@mao-liu how about this? 
https://github.com/intelematics/bricklayer/releases

you can now run 
```
pip install https://github.com/intelematics/bricklayer/releases/download/0.0.6/bricklayer-0.0.6-py3-none-any.whl
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on pull request merged into master branch events
+  push:
+    branches: [ master ]
+
+  pull_request:
+    branches: [ master ]
+
```
Comment (mao-liu) [inline .github/workflows/main.yml:13]:
```suggestion
```

**Review by mao-liu — APPROVED** [review]
🎉

**Review by ben-thwaites — APPROVED** [review]

**Pull request merged**